src/components/FlexibleForm/ControlDecorators/withReferenceValues/withReferenceValues.js

```diff
@@ -55,8 +57,48 @@ export const withReferenceValues = memo(props => {
     }
   };
 
+  const validateQuotedText = value => {
+    let isValid = true;
+    const pattern = /"[^"]+"/g;
+    const matches = value.match(pattern);
+
+    if (matches) {
+      const dataSet = dataOptions.map(option => option.name);
+
+      for (const str of matches) {
+        const croppedStr = str.slice(1, -1);
+
+        isValid = dataSet.some(data => data === croppedStr);
```

Comment:
I don't think that you need `dataSet` here, you can use destructuring.

`isValid = dataOptions.some(({ name }) => name === croppedStr);`